### PR TITLE
feature: make get-ticket just a way to use get

### PR DESCRIPTION
**Pull request by rklaehn**
you can either use get --ticket <ticket> or get --peer..., but it is all handled by the get subcommand.

The args are configured so that you get decent error messages when misusing. E.g.

```
❯ cargo run get --ticket ebim463bsc4cl3t5vkn7vlacngxvu5wkv5o3fzj5psw2whnxyipfeigxxvopirxcngbsl3eol4a5hchk7tpvzdcwc6qy42equqon2n2w7yaagacsjt2rhzg3amambkabqtsnwayaycuadc7e3mbq --peer 266vz5cg4juygjpmrzpqdu4i5l6n6xemkyl2ddtiscsbzxjxk37a
error: the argument '--ticket <TICKET>' cannot be used with '--peer <PEER>'
```

The netsim test failure is because netsim is using iroh get-ticket instead of iroh get --ticket. I got a branch of chuck where this is corrected. Just replace `get-ticket` with `get --ticket`

### CODE REVIEW

**Comment by b5** [comment]
love it. This is touching some of the same code as #1159, which if it's cool with you @rklaehn, I'd like to merge first.

**Comment by rklaehn** [comment]
> love it. This is touching some of the same code as #1159, which if it's cool with you @rklaehn, I'd like to merge first.

You go ahead. This is tiny, so I can deal with the fallout...

File: iroh/src/commands.rs [hunk 1]
```diff
@@ -194,10 +190,16 @@ pub enum Commands {
...
+        #[clap(conflicts_with = "ticket", required_unless_present = "ticket")]
+        hash: Option<Blake3Cid>,
         /// PeerId of the provider
-        #[clap(long, short)]
-        peer: PeerId,
+        #[clap(
+            long,
+            short,
+            conflicts_with = "ticket",
+            required_unless_present = "ticket"
```
Comment (flub) [inline iroh/src/commands.rs:200]: whoa, this allows you to do complex things.  nice

**Review by flub — APPROVED** [review]

**Comment by dignifiedquire** [comment]
merge conflicts detected 😬

**Review by dignifiedquire — APPROVED** [review]

**Comment by dignifiedquire** [comment]
This finishes https://github.com/n0-computer/iroh/issues/1163

File: iroh/src/commands.rs (hunk 1 above)
Comment (rklaehn) [reply iroh/src/commands.rs:200]: I had chatgpt explain it to me. There are still some downsides, e.g. the usage message is a bit weird.

**Review by dignifiedquire — APPROVED** [review]

**Pull request merged**